stsx

```diff
@@ -28,67 +28,114 @@
 
-#
-# Args analysis
-#
-
-method="$1"
-watchdog="$2 $3"
-maxnum="$4"
-if [ "x$1" = x ] ; then
-  cat 1>&2 <<"EOUSAGE"
-Usage: $0 <s4|s4k|s3|s3k|s0|s0k|s4simdev|s4simcore|s3simdev|s3simplatform|s3simcore|s4simcpu> [watchdog port [maxnum]]
+-m    Specifies the maximum of retries.
+-l    Specifies the watchdog server.
+      Example: localhost
+-p    Specifies the port of the watchdog server
+      Default: 1234
 
 Possible deductions:
 simdev fails                 -> no BIOS issue
 simcore passes AND s4k fails -> BIOS issue
-s4simcpu fails		     -> no BIOS issue + CPU hotplug bug
+s4simcpu fails               -> no BIOS issue + CPU hotplug bug
 
 Only use coresim on machines that switch off power to busses in ACPI.
 
-EOUSAGE
+EOF
+}
+
+error(){
+  echo $1
+  echo ""
+  usage
   exit 1
-fi
-if [ "x$watchdog" = "x " ] ; then
-  watchdog=""
-fi
-if [ "x$maxnum" = x ] ; then
-  maxnum=9999999999
+}
+
+if [ x`id -u` != x0 ] ; then
+  error "Please call the script as root!"
 fi
 
+#
+# Args analysis
+#
+
+# Default arguments
+method=""
+watchdog=""
+watchdog_port=1234
+maxnum=9999999999
+
+while getopts "l:m:p:h" opt; do
+  case ${opt} in
+    m)
+      maxnum=$OPTARG
+      ;;
+    l)
+      watchdog=$OPTARG
+      ;;
+    p)
+      watchdog_port=$OPTARG
+      ;;
+    h)
+      usage
+      exit 0
+      ;;
+    \?)
+      error "Unknow option!"
+      ;;
+  esac
+done
+shift $(($OPTIND-1))
+method="$1"
+
 delay=10
 resume=""
 alternate=0
 simulation="none"
 
 case "x$method" in
-xs4)		resume=70;		;;
-xs4k)		resume=70;		;;
-xs3)		resume=20;		;;
-xs3k)		resume=20;		;;
-xs0)    resume=20;    ;;
-xs0k)   resume=20;    ;;
-xs43)		alternate=1;		method="s4"	resume=70	;;
-xs43k)          alternate=1;      	method="s4"	resume=70	;;
-xs4simdev)	simulation="devices";	delay=5;	;;
-xs4simcore)	simulation="core";	delay=5;	;;
-xs3simdev)	simulation="devices";	delay=5;	;;
-xs3simplatform) simulation="platform";  delay=5;	;;
-xs3simcore)	simulation="core";	delay=5;	;;
-xs4simcpu)	simulation="processors";delay=0;	;;
-*)
-  echo "unknown method $method"
-  exit 1
+  xs4)            resume=70 ;;
+  xs4k)           resume=70 ;;
+  xs3)            resume=20 ;;
+  xs3k)           resume=20 ;;
+  xs0)            resume=20; ;;
+  xs0k)           resume=20; ;;
+  xs43)           alternate=1 method="s4" resume=70 ;;
+  xs43k)          alternate=1 method="s4" resume=70 ;;
+  xs4simdev)      simulation="devices" delay=5 ;;
+  xs4simcore)     simulation="core" delay=5 ;;
+  xs3simdev)      simulation="devices" delay=5 ;;
+  xs3simplatform) simulation="platform" delay=5 ;;
+  xs3simcore)     simulation="core" delay=5; ;;
+  xs4simcpu)      simulation="processors" delay=0 ;;
+  x)              error "Argument missing. Please specify a method.";;
+  *)              error "Unknown method $method"
 esac
 
+
+if ! [ "$watchdog_port" -eq "$watchdog_port" ] 2>/dev/null;then
+  error "Invalid port number"
+fi
+
+if ! [ "$watchdog_port" -ge 1 -a "$watchdog_port" -le 65535 ]; then
```

Comment:
Don't understand this line. This will always evaluate to false, right? What's the purpose behind this if-clause?